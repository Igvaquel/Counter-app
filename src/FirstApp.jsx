import PropTypes from "prop-types"

export const FirstApp = ( { title, subTitle } ) => {
    return (
        <>
            <h1 data-testid="test-title" >{ title }</h1>
            <h3>{ subTitle }</h3>

        </>
    )
}

FirstApp.propTypes = {
    title: PropTypes.string.isRequired,
    subTitle: PropTypes.string
}

FirstApp.defaultPorps = {
    title: 'No hay titulo',
    subTitle: 'No hay subtitulo'
}