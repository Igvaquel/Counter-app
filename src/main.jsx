import React from 'react'
import ReactDOM from 'react-dom/client' 
import "./styles.css"
import { FirstApp } from './FirstApp'
import { CounterApp } from './CounterApp'

ReactDOM.createRoot( document.getElementById('root') ).render(
    <React.StrictMode>
        <CounterApp value={0} />
        {/* <FirstApp title='Hola'
        subTitle="Mundo"/> */}
    </React.StrictMode>
)