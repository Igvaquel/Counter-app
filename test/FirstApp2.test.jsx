import { getByTestId, render, screen } from "@testing-library/react"
import { FirstApp } from "../src/FirstApp"

describe('pruebas en <FirstApp />', () => { 
    
    const title = "Hola soy ignacio"
    const subTitle = "subtitulo"

   test('debe de hacer match con el snapshot', () => { 
        
        const { container } = render( <FirstApp title={title}/> )
        expect( container ).toMatchSnapshot()

   })

    test('debe demostrar el mensaje "Hola soy ignacio"', () => { 
        
        render( <FirstApp title={title}/> )
        expect( screen.getByText(title)).toBeTruthy()

    })
    test('debe demostrar el titulo en un h1', () => { 
        
        render( <FirstApp title={title}/> )

        expect( screen.getByRole('heading', {level: 1}).innerHTML ).toContain( title )

    })

    test('debe de mostrar el subtitulo enviado por props', () => { 
        
        render( <FirstApp title={title} subTitle={subTitle}/> )

        expect( screen.getAllByText(subTitle).length).toBe(1)

     })
 })