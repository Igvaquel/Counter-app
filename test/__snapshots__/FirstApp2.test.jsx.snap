// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en <FirstApp /> debe de hacer match con el snapshot 1`] = `
<div>
  <h1
    data-testid="test-title"
  >
    Hola soy ignacio
  </h1>
  <h3 />
</div>
`;
